src/util/Random.c

```diff
@@ -34,29 +58,24 @@ void random_mpzOfLength(mpz_t result, const unsigned int numberOfBits)
 
 void random_mpzInRange(mpz_t result, const mpz_t range)
 {
-    mpz_t x;
-    mpz_init(x);
-
-    random_mpzOfLength(x, mpz_sizeinbase(range, 2));
-
-    mpz_mod(result, x, range);
-
-    mpz_clear(x);
+    do
+    {
+        random_mpzOfLength(result, mpz_sizeinbase(range, 2));
```

Comment:
Just as `rangeBitLength` is calculated before the loop, we should use the same approach here in the case of  `mpz_sizeinbase`, right? Minor thing, most probably optimized away by the compiler, but might be worth changing.

Reply:
Good call, done.